ExampleApp/equityTicketConfig.json

```diff
@@ -0,0 +1,195 @@
+{
+	"fields": [
+		{
+			"apiParameter": "token",
+			"inputType": "hidden"
+		},
+		{
+			"apiParameter": "accountNumber",
+			"inputType": "accountSelection"
+		},
+		{
+			"apiParameter": "orderAction",
+			"label": "Action",
+			"inputType": "options",
+			"options": [
+				{
+					"value": "buy",
+					"label": "Buy"
+				},
+				{
+					"value": "sell",
+					"label": "Sell"
+				}
+			]
+		},
+		{
+			"apiParameter": "orderPrice",
+			"label": "Price",
+			"inputType": "options",
+			"options": [
+				{
+					"value": "market",
+					"label": "Market"
+				},
+				{
+					"value": "limit",
+					"label": "Limit"
+				},
+				{
+					"value": "stopLimit",
+					"label": "Stop Limit"
+				},
+				{
+					"value": "stop",
+					"label": "Stop"
+				}
+			]
+		},
+		{
+			"apiParameter": "orderQuantityType",
+			"label": "Quantity Type",
+			"inputType": "options",
+			"options": [
+				{
+					"value": "SHARES",
+					"label": "Shares"
+				},
+				{
+					"value": "TOTAL_VALUE",
+					"label": "USD",
+					"dependencies": [
```

Comment:
1. I'm not hot on `dependencies`. Was this inspired by JSON Schema? `prerequisites` seems to carry more meaning.
2. It is unclear to me how we are logically chaining together `dependencies`. My initial take was that these were `&&`’d together, however, as there are recurrent entries with the same `criteria` it would seem we are `||`’ing them. Perhaps we don’t need both operator types in the present moment, however, the cost of doing it later on could be quite high — and as I said it’s not obvious in every payload that these are `||`s to start with.

Perhaps we add reformat to the following:
```json
"dependencies": {
   "operator": "||"          // can be || or &&, ignore in the case of a 
   "conditions": [
       {
          "operator": "==",
			"value": "limit",
			"criteria": "orderPrice"
       }, 
       {
			"operator": "==",
			"value": "stopLimit",
			"criteria", "orderPrice"
       }
    ]
}
```

Thinking outward, it would also be possible to nest conditions by supplying `||` or `&&`  along with a list of `conditions` within a specific condition.

Reply:
Yeah I had thought about this too but at that point do we make it totally generic and allow recursive conditions? Will that drastically complicate a new integration (maybe unnecessarily)?

Reply:
If we only implement one layer deep comparisons for the time being the same general schema could be expanded upon later, whereas this current schema require a rehaul. The client code will still need to be updated, however, I don't see how this is any different than the fact we supply `condition` but won't be providing every possible condition (`>`, `<`, etc...) in V1.

Reply:
Related to the logical operators, this is what MongoDB do: 
```
   $and : [
        { $or : [ { price : 0.99 }, { price : 1.99 } ] },
        { $or : [ { sale : true }, { qty : { $lt : 20 } } ] }
    ]
```
And this is DB2:
`{"$or":[{"age":4},{"name":"Joe"}]}`

These kind of formats allows for a lot of flexibility but I think It might be overkill for our needs.